python/setup.py

```diff
@@ -271,6 +268,12 @@ def get_outputs(self):
     version_tag = list(tree.getroot().findall('{http://maven.apache.org/POM/4.0.0}version'))[0]
     os.environ["SETUPTOOLS_SCM_PRETEND_VERSION"] = version_tag.text.replace("-SNAPSHOT", "a0")
 
+long_description = """Apache Arrow is a columnar in-memory analytics layer
+designed to accelerate big data. It houses a set of canonical in-memory
+representations of flat and hierarchical data along with multiple
+language-bindings for structure manipulation. It also provides IPC
+and common algorithm implementations."""
```

Comment:
We should go through an iteration on this description, also for the main site. On the top of GitHub it would be good to state that instead of "It also provides IPC and common algorithm implementations" instead sometime more descriptive like "It provides high performance streaming and batch messaging tools"